e2e/upgrade-testing.go

```diff
@@ -0,0 +1,76 @@
+package e2e
+
+import (
+	"fmt"
+	"os"
+	"os/exec"
+	"strconv"
+	"strings"
+)
+
+// upgradeCSI deploys a desired ceph-csi release version.
+func upgradeCSI(version string) error {
+	tempDir := "/tmp/csi"
+	gitRepo := "https://github.com/ceph/ceph-csi.git"
+	// clone the desired release branch inside a temporary directory.
+	cmd := exec.Command("git", "clone", "--single-branch", "--branch", version, gitRepo, tempDir)
+	cmd.Stderr = os.Stderr
+	cmd.Stdout = os.Stdout
+	if err := cmd.Run(); err != nil {
+		return fmt.Errorf("unable to clone repo %s : %w", gitRepo, err)
+	}
+	err := os.Chdir(tempDir + "/e2e")
+	if err != nil {
+		return fmt.Errorf("unable to switch directory : %w", err)
+	}
+	return nil
+}
+
+// upgradeAndDeployCSI upgrades the CSI to a specific release.
+func upgradeAndDeployCSI(version, testtype string) error {
+	err := upgradeCSI(version)
+	if err != nil {
+		return fmt.Errorf("failed to upgrade driver %w", err)
+	}
+	skipPSP, err = checkIfPSPDeploymentRequired(version)
+	if err != nil {
+		return fmt.Errorf("failed to validate PSP requirement %w", err)
+	}
+	switch testtype {
+	case "cephfs":
+		deployCephfsPlugin()
```

Comment:
you need to update the `rbdDirPath` and cepfsDirPath` with new paths to deploy templates

Reply:
I currently go to the directory itself(and come back) where we deploy the upgrade-version branch so that no path needs to be changed [here](https://github.com/ceph/ceph-csi/pull/1243/files/f15281e4abfedb8a15aee93a08699c5203aa2a2a)... I can change it too if you suggest?